src/lib.rs

```diff
@@ -260,26 +293,36 @@ impl Watch {
     }
 }
 
+#[non_exhaustive]
 #[derive(Debug, StructOpt)]
 pub struct DevServer {
     #[structopt(long, default_value = "127.0.0.1")]
-    ip: IpAddr,
+    pub ip: IpAddr,
     #[structopt(long, default_value = "8000")]
-    port: u16,
+    pub port: u16,
 
     #[structopt(flatten)]
-    watch: Watch,
+    pub watch: Watch,
+    #[structopt(skip)]
+    pub served_path: Option<PathBuf>,
+    #[structopt(skip)]
+    pub release: bool,
 }
 
 impl DevServer {
-    pub fn serve(&self, build_dir_path: impl AsRef<Path>) -> Result<()> {
+    pub fn serve(self) -> Result<()> {
```

Comment:
1. Let's move the content of `serve()` to a private function.
2. Combine function `serve()` and `serve_and_watch()` and rename to `start()` ("dev server start" is common language for that)
3. Move the `command` argument to an optional field and use the builder pattern
4. Do not start the watch loop if the command argument is None
5. Avoid cloning: remove Clone wherever you added it (Watch)

Reply:
You did not respect the point 3. Now the watch parameters are not accessible to the end user.